src/java.logging/share/classes/java/util/logging/FileHandler.java

```diff
@@ -180,18 +190,26 @@ private static final class MeteredStream extends OutputStream {
         public void write(int b) throws IOException {
             out.write(b);
             written++;
+            flushOrRotateIfFull();
```

Comment:
I don't think that's correct. You don't want to flush and rotate in the middle of publishing a LogRecord. You want to flush and rotate either before or after - but not in the middle.

Reply:
This sounds fragile to me. I'd rather use a package private method called within the lock to signal that a record has been written.